pkg/controller/volume/attachdetach/util/util.go

```diff
@@ -397,3 +397,14 @@ func isCSIMigrationSupportedOnNode(nodeName types.NodeName, spec *volume.Spec, v
 
 	return isMigratedOnNode, nil
 }
+
+// NodeHasTaint returns true if the node has the taint passed in the argument
+// ToDo: Add unit test
+func NodeHasTaint(taint string, nodeObject v1.Node) bool{
```

Comment:
Do we already have some util function like this in code base?

Reply:
Thinking to move here 
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/node/node.go
And the name could `HasTaintByKey`
wdyt?

Reply:
See here:
https://github.com/kubernetes/kubernetes/blob/v1.24.0-alpha.3/pkg/util/taints/taints.go#L269